tools/lsp/language.rs

```diff
@@ -1205,26 +1283,96 @@ fn get_code_lenses(
     document_cache: &mut DocumentCache,
     text_document: &lsp_types::TextDocumentIdentifier,
 ) -> Option<Vec<CodeLens>> {
-    if cfg!(any(feature = "preview-builtin", feature = "preview-external")) {
-        let filepath = uri_to_file(&text_document.uri)?;
-        let doc = document_cache.documents.get_document(&filepath)?;
+    let filepath = uri_to_file(&text_document.uri)?;
+    let doc = document_cache.documents.get_document(&filepath)?;
+    let version = document_cache.document_version(&text_document.uri);
 
-        let inner_components = doc.inner_components.clone();
+    let mut result = vec![];
 
-        let mut r = vec![];
+    if cfg!(any(feature = "preview-builtin", feature = "preview-external")) {
+        let inner_components = doc.inner_components.clone();
 
         // Handle preview lens
-        r.extend(inner_components.iter().filter(|c| !c.is_global()).filter_map(|c| {
+        result.extend(inner_components.iter().filter(|c| !c.is_global()).filter_map(|c| {
             Some(CodeLens {
                 range: util::map_node(&c.root_element.borrow().debug.first()?.0)?,
                 command: Some(create_show_preview_command(true, &text_document.uri, c.id.as_str())),
                 data: None,
             })
         }));
+    }
 
-        Some(r)
-    } else {
+    if let Some(node) = &doc.node {
+        if let Some(range) = util::map_node(node) {
+            let has_non_ws_token = node
+                .children_with_tokens()
+                .any(|nt| nt.kind() != SyntaxKind::Whitespace && nt.kind() != SyntaxKind::Eof);
```

Comment:
Should we also accept a comment?  (eg license header comments)

Reply:
I was wondering the same, but the "empty page" effect is already broken at the point where you have a comment already ;-)

So far it just replaces the entire string with an "OK" formatted default text. If we preserve comments, then we need to be way more clever than that to integrate the comment properly.